src/coreclr/src/vm/nativeimage.h

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _NATIVEIMAGE_H
+#define _NATIVEIMAGE_H
+
+#include "readytoruninfo.h"
+
+/// <summary>
+/// This structure is used in NativeImage to map simple names of component assemblies
+/// to their indices within the component assembly header table.
+/// </summary>
+struct AssemblyNameIndex
+{
+    SString Name;
+    int32_t Index;
+    
+    AssemblyNameIndex() : Index(-1) {}
+    AssemblyNameIndex(const SString& name, int32_t index);
+};
+
+class AssemblyNameIndexHashTraits : public NoRemoveSHashTraits< PtrSHashWithCleanupTraits<AssemblyNameIndex, const SString&> >
+{
+public:
+    static AssemblyNameIndex *Null() { return NULL; }
+    static bool IsNull(const AssemblyNameIndex *e) { return e == NULL; }
+
+    static const SString& GetKey(const AssemblyNameIndex *assemblyNameIndex) { return assemblyNameIndex->Name; }
+    static BOOL Equals(const SString& a, const SString& b) { return a.Equals(b); }
+    static count_t Hash(const SString& a) { return a.Hash(); }
+};
+
+class AssemblyLoadContext;
+class ReadyToRunInfo;
+class PEFile;
+class PEImage;
+
+class NativeImage
```

Comment:
A comment explaining what this is representing may be useful

Reply:
Added explanatory comment in 9th commit.